envconfig.go

```diff
@@ -114,12 +116,12 @@ func gatherInfo(prefix string, spec interface{}) ([]varInfo, error) {
 			}
 		}
 		if info.Alt[0] != "" {
-			info.Key = info.Alt[0]
+			info.Key = strings.Replace(info.Alt[0], "__", "_", -1)
```

Comment:
looks like you should have filtered some empty strings before concatening string (e.g. prefix + "_" + "")